Sources/Verge/Store/Pipeline.swift

```diff
@@ -86,8 +86,7 @@ public enum Pipelines {
       }
       
       guard
-        previous.primitive == input.primitive ||
-          previous.primitive[keyPath: keyPath] == input.primitive[keyPath: keyPath]
+        previous.primitive[keyPath: keyPath] == input.primitive[keyPath: keyPath]
```

Comment:
should not comparing data source, as much bigger cost than comparing selected values.